File: modules/_foundation/src/main/java/com/projecthub/base/core/event/DomainEvent.java
package com.projecthub.base.core.event;

import java.time.Instant;

/**
 * Base interface for all domain events in the system.
 * Contains common properties that all domain events should have.
 */
public interface DomainEvent {
    
    /**
     * Gets the unique identifier for this event.
     * 
     * @return The event ID
     */
    String getEventId();
    
    /**
     * Gets the timestamp when this event occurred.
     * 
     * @return The event timestamp
     */
    Instant getTimestamp();
    
    /**
     * Gets the type of this domain event, used for routing and handling.
     * 
     * @return The event type
     */
    String getEventType();
}

File: src/main/kotlin/com/projecthub/domain/event/DomainEvent.kt
package com.projecthub.domain.event

import java.time.Instant
import java.util.UUID

/**
 * Base class for all domain events in the system.
 * Domain events represent significant state changes within a domain context.
 */
abstract class DomainEvent {
    val eventId: String = UUID.randomUUID().toString()
    val timestamp: Instant = Instant.now()

    /**
     * Unique event type identifier, typically the simple class name
     */
    val eventType: String = this.javaClass.simpleName
}
